server/stt/io/feeding_services/stt_tt_content_api.py

```diff
@@ -0,0 +1,203 @@
+            "required": True,
+        },
+        {
+            "id": "api_key",
+            "type": "text",
+            "label": "API Key (ApiKey or raw token)",
+            "placeholder": "ApiKey <token> OR raw <token>",
+            "required": True,
+        },
+        {
+            "id": "page_size",
+            "type": "text",
+            "label": "Page size (s)",
+            "placeholder": "50",
+            "required": False,
+            "default": "50",
+            "description": (
+                "Number of items to request per page from TT Content API "
+                "(query param s). Range: 1-1000."
+            ),
+        },
+        {
+            "id": "max_pages",
+            "type": "text",
+            "label": "Max pages",
+            "placeholder": "200",
+            "required": False,
+            "default": "200",
+            "description": (
+                "Safety cap for the number of pages to fetch when "
+                "paginating. Range: 1-10000."
+            ),
+        },
+        {
+            "id": "timeout",
+            "type": "text",
+            "label": "Request timeout (seconds)",
+            "placeholder": "300",
+            "required": False,
+            "default": "300",
+            "description": (
+                "HTTP request timeout per page when calling TT Content API."
+            ),
+        },
+    ]
+
+    def _update(self, provider, update) -> Iterable[Dict]:
+        """
+        TT-specific update that fetches all pages and yields parsed dict items.
+        """
+        json_items = self._fetch_tt_data(provider)
+        if not isinstance(json_items, list):
+            json_items = [json_items]
+
+        parsed_items: List[Dict] = []
+
+        # Resolve parser once
+        parser = self.get_feed_parser(provider)
+
+        for item in json_items:
+            try:
+                if not isinstance(item, dict):
+                    continue
+
+                parsed_result = parser.parse(item, provider)
+                # Only return dict items to the ingest pipeline
+                if isinstance(parsed_result, list):
+                    parsed_items.extend(
+                        [x for x in parsed_result if isinstance(x, dict)]
+                    )
+                elif isinstance(parsed_result, dict):
+                    parsed_items.append(parsed_result)
+                else:
+                    pass
+            except Exception as ex:
+                logger.error("Error processing item: %s", str(ex))
+                raise ParserError.parseMessageError(ex, provider, data=item)
+
+        # Final guard: ensure only dicts are returned
+        # (avoids filter_expired_items crash)
+        parsed_items = [it for it in parsed_items if isinstance(it, dict)]
+        return parsed_items
+
+    def _fetch_tt_data(self, provider) -> List[Dict]:
+        """
+        Fetch all items from TT Content API with pagination.
```

Comment:
I wonder why actually fetching all items every time it runs, you could use the `trs` param with last updated timestamp to only get new content since last time